### DISCUSSION TITLE
MV2 closeout imminent. Does uBOL has legacy code that rely on MV2?

### DISCUSSION THREAD

**Discussion by Meyers07** [uBlockOrigin/uBOL-home#85] [Q&A] [answered]
Been using MV2 since the first announcement back in 2022 (around September and October) and i am satisfied on how it basically blocks the same and act the same though the UI is far more basic, inability to see which ad domains are blocked and there are no filters apart from the ones that came with the app. None of it was an issue to users and one could add a secure DNS like Adguard or Pihole to complement.

One question, does uBOL has legacy fallback codes using MV2 because it perform so well? And Google gave the go ahead to shut down MV2 as of yesterday.

**Comment by MasterKia** [comment]
> because it perform so well?

https://github.com/uBlockOrigin/uBOL-home/wiki/Frequently-asked-questions-(FAQ)#is-ubol-more-efficient-cpu--and-memory-wise-than-ubo

**Reply by Meyers07** [reply]
i already understand the noted issues but it basically perform just as good as UBO for common users. My real question only if is it have MV2 reliant fallback or legacy code or not, because during the cutoff that will be affected.

**Reply by MasterKia** [reply, marked as answer]
https://github.com/uBlockOrigin/uBOL-home#description:
> uBOL is a permission-less **MV3 API-based** content blocker.

It's all based on manifest v3.

> MV2 reliant fallback or legacy code 

There's no such thing as MV2 fallback code in a MV3 extension, where did you get this idea from?

**Reply by Meyers07** [reply]
As a common user yet knowing the impact of MV2 shutoff due to using PC browsers and extensions for almost 20 years i am predicting or just asking if there's a "just-in-case" measure especially how UBOL performs just as good as predecessors with (for common user) negligible drawbacks.
Well then i consider the issue closed.

**Reply by gorhill** [reply]
See <https://github.com/uBlockOrigin/uBOL-home/wiki/Frequently-asked-questions-(FAQ)>
